generators/client-2/templates/src/main/webapp/app/_app.main.ts

```diff
@@ -1,4 +1,11 @@
+import { UpgradeAdapterRef } from "@angular/upgrade";
 import { upgradeAdapter } from './upgrade_adapter';
 import './app.module';
 
-upgradeAdapter.bootstrap(document.body, ['<%=angularAppName%>.app'], {strictDi: true});
+function readyFn(ref: UpgradeAdapterRef) {
+    let $urlRouter = ref.ng1Injector.get('$urlRouter');
+    $urlRouter.listen();
+    $urlRouter.sync();
+}
+
+upgradeAdapter.bootstrap(document.body, ['<%=angularAppName%>.app'], {strictDi: true}).ready(readyFn);
```

Comment:
ng1/ng2 bootstrap timing can be tricky.
Elsewhere, we do `$urlRouterProvider.deferIntercept()` to defer the initial url sync. Here, we know ng2 is ready, so we tell the router to sync.